src/main/java/htsjdk/samtools/BamFileIoUtils.java

```diff
@@ -65,58 +79,61 @@ public static void reheaderBamFile(final SAMFileHeader samFileHeader, final File
             // a) It's good to check that the end of the file is valid and b) we need to know if there's a terminator block and not copy it if skipTerminator is true
             final BlockCompressedInputStream.FileTermination term = BlockCompressedInputStream.checkTermination(inputFile);
             if (term == BlockCompressedInputStream.FileTermination.DEFECTIVE)
-                throw new SAMException(inputFile.getAbsolutePath() + " does not have a valid GZIP block at the end of the file.");
+                throw new SAMException(inputFile.toUri() + " does not have a valid GZIP block at the end of the file.");
 
             if (skipHeader) {
                 final long vOffsetOfFirstRecord = SAMUtils.findVirtualOffsetOfFirstRecordInBam(inputFile);
-                final BlockCompressedInputStream blockIn = new BlockCompressedInputStream(inputFile);
-                blockIn.seek(vOffsetOfFirstRecord);
-                final long remainingInBlock = blockIn.available();
-
-                // If we found the end of the header then write the remainder of this block out as a
-                // new gzip block and then break out of the while loop
-                if (remainingInBlock >= 0) {
-                    final BlockCompressedOutputStream blockOut = new BlockCompressedOutputStream(outputStream, (Path)null);
-                    IOUtil.transferByStream(blockIn, blockOut, remainingInBlock);
-                    blockOut.flush();
-                    // Don't close blockOut because closing underlying stream would break everything
-                }
 
-                long pos = BlockCompressedFilePointerUtil.getBlockAddress(blockIn.getFilePointer());
-                blockIn.close();
-                while (pos > 0) {
-                    pos -= in.skip(pos);
+                // tsato: curious --- why do we need BlockCompressedInputStream at all here?
+                try (final BlockCompressedInputStream blockIn = new BlockCompressedInputStream(inputFile)) {
+                    blockIn.seek(vOffsetOfFirstRecord);
+                    final long remainingInBlock = blockIn.available();
+
+                    // If we found the end of the header then write the remainder of this block out as a
+                    // new gzip block and then break out of the while loop (tsato: update this comment)
+                    if (remainingInBlock >= 0) {
+                        final BlockCompressedOutputStream blockOut = new BlockCompressedOutputStream(outputStream, (Path) null);
+                        IOUtil.transferByStream(blockIn, blockOut, remainingInBlock);
+                        blockOut.flush();
+                        // Don't close blockOut because closing underlying stream would break everything
+                    }
+
+                    final long pos = BlockCompressedFilePointerUtil.getBlockAddress(blockIn.getFilePointer());
+                    blockIn.close(); // tsato: why doesn't IntelliJ say this is unnecessary?
```

Comment:
You could leave out that close.  I don't know why intellij isn't telling you that but it looks like it's redundant to me.